### ISSUE TITLE
Microsoft authentification error 2148916238

### ISSUE THREAD

**Issue by IbanEtchep** [Azuriom/Azuriom#256]
### Bug description

Hello, some of my users can't connect to the website with their microsoft account.

The error code is: 2148916238
https://wiki.vg/Microsoft_Authentication_Scheme :
2148916238: The account is a child (under 18) and cannot proceed unless the account is added to a Family by an adult. This only seems to occur when using a custom Microsoft Azure application. When using the Minecraft launchers client id, this doesn't trigger.

### Step to reproduce

Connect with a child account

### Errors / Logs
```
[2022-11-11 13:44:24] production.ERROR: Invalid Xbox response: 2148916238 {"exception":"[object] (Laravel\\Socialite\\Two\\InvalidStateException(code: 0): Invalid Xbox response: 2148916238 at /var/www/vhosts/spartacube.fr/httpdocs/app/Socialite/Xbox/XboxProvider.php:89)
[stacktrace]
#0 /var/www/vhosts/spartacube.fr/httpdocs/app/Socialite/Minecraft/MinecraftProvider.php(19): Azuriom\\Socialite\\Xbox\\XboxProvider->getXboxIdentityToken()
#1 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/socialite/src/Two/AbstractProvider.php(243): Azuriom\\Socialite\\Minecraft\\MinecraftProvider->getUserByToken()
#2 /var/www/vhosts/spartacube.fr/httpdocs/app/Http/Controllers/Auth/LoginController.php(121): Laravel\\Socialite\\Two\\AbstractProvider->user()
#3 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): Azuriom\\Http\\Controllers\\Auth\\LoginController->handleProviderCallback()
#4 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(43): Illuminate\\Routing\\Controller->callAction()
#5 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Routing/Route.php(260): Illuminate\\Routing\\ControllerDispatcher->dispatch()
#6 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\\Routing\\Route->runController()
#7 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Routing/Router.php(725): Illuminate\\Routing\\Route->run()
#8 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
#9 /var/www/vhosts/spartacube.fr/httpdocs/app/Http/Middleware/RedirectIfAuthenticated.php(30): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#10 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Azuriom\\Http\\Middleware\\RedirectIfAuthenticated->handle()
#11 /var/www/vhosts/spartacube.fr/httpdocs/plugins/invoicepro/src/Middleware/CheckForBilling.php(26): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#12 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Azuriom\\Plugin\\Invoicepro\\Middleware\\CheckForBilling->handle()
#13 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#14 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle()
#15 /var/www/vhosts/spartacube.fr/httpdocs/app/Http/Middleware/LogoutIfSuspended.php(26): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#16 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Azuriom\\Http\\Middleware\\LogoutIfSuspended->handle()
#17 /var/www/vhosts/spartacube.fr/httpdocs/app/Http/Middleware/CheckForMaintenanceSettings.php(45): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#18 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Azuriom\\Http\\Middleware\\CheckForMaintenanceSettings->handle()
#19 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#20 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle()
#21 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Session/Middleware/AuthenticateSession.php(40): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#22 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Session\\Middleware\\AuthenticateSession->handle()
#23 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#24 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle()
#25 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#26 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest()
#27 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Session\\Middleware\\StartSession->handle()
#28 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#29 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle()
#30 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#31 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle()
#32 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#33 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Routing/Router.php(726): Illuminate\\Pipeline\\Pipeline->then()
#34 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Routing/Router.php(703): Illuminate\\Routing\\Router->runRouteWithinStack()
#35 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Routing/Router.php(667): Illuminate\\Routing\\Router->runRoute()
#36 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Routing/Router.php(656): Illuminate\\Routing\\Router->dispatchToRoute()
#37 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(190): Illuminate\\Routing\\Router->dispatch()
#38 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
#39 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#40 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#41 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle()
#42 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#43 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#44 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
#45 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#46 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#47 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#48 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
#49 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Http/Middleware/HandleCors.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#50 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Http\\Middleware\\HandleCors->handle()
#51 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#52 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Http\\Middleware\\TrustProxies->handle()
#53 /var/www/vhosts/spartacube.fr/httpdocs/app/Http/Middleware/EnsureInstalled.php(24): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#54 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Azuriom\\Http\\Middleware\\EnsureInstalled->handle()
#55 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#56 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(165): Illuminate\\Pipeline\\Pipeline->then()
#57 /var/www/vhosts/spartacube.fr/httpdocs/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(134): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#58 /var/www/vhosts/spartacube.fr/httpdocs/public/index.php(52): Illuminate\\Foundation\\Http\\Kernel->handle()
#59 /var/www/vhosts/spartacube.fr/httpdocs/server.php(19): require_once('...')
#60 {main}
```

### Web server

Apache

### Database type

MariaDB

### PHP version

8.1

**Comment by MrMicky-FR** [comment]
Hum not sure how to fix this one

**Comment by IbanEtchep** [comment]
Peut être juste mettre un message quand il y a ce message disant qu'il ne peut pas se connecter à cause du contrôle parentale.

**Comment by MrMicky-FR** [comment]
Fixed in 126e89e6dcac0cdff3f3d35467efa3fa35b52163
